### PR TITLE
State Machine

### PR DESCRIPTION
**Pull request by simo9000**
I went ahead and further enhanced queue to include a state machine function for proceeding through a series of alerts. Instead of just handing the the steps function an integer like in dynamic queuing, i made a flow control object that allows the steps function to control state with helper functions. I made another [fiddle](https://jsfiddle.net/simo9000/1rs7n2td/2/) to show this in action. If you pick the right drop downs you can go in an infinite loop. Again if this is in the right direction let me know and i'll do more tests and docs.

### CODE REVIEW

**Comment by limonte** [comment]
@simo9000 looks good for me :+1: please go ahead with tests and docs.

File: test/tests.js [hunk 1]
```diff
@@ -230,6 +230,88 @@ test('dynamic queue', function(assert) {
   });
 });
 
+test('state machine', function(assert) {
```
Comment (simo9000) [inline test/tests.js]: I am scratching my head why houndci-bot flagged this. The linter didn't have a problem with it.

Comment (limonte) [inline test/tests.js]: it's not important, just ignore it

**Comment by limonte** [comment]
Thank you @simo9000 for adding docs and tests. Could you please figure out why tests are failing?

**Comment by simo9000** [comment]
Yeah i saw that. I've bogged down with something else after I pushed
earlier. The strange thing is that all the tests passed when I ran them
locally. I might not get a chance to till tomorrow to figure this out.

On Aug 27, 2016 4:16 PM, "Limon Monte" <notifications@github.com> wrote:

Thank you @simo9000 <https://github.com/simo9000> for adding docs and
tests. Could you please figure out why tests are failing?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/limonte/sweetalert2/pull/241#issuecomment-242946209>,
or mute the thread

File: test/tests.js [hunk 2]
```diff
@@ -230,6 +230,88 @@ test('dynamic queue', function(assert) {
...
+        };
+      default:
+        return obj.current;
+    }
+  };
+  swal.queue(stepGen, ['Step 1', 'Step 5']);
+  assert.equal('Step 1', $('.swal2-modal h2').text());
+  swal.clickConfirm();
+  setTimeout(function() {
+    assert.equal('Step 2', $('.swal2-modal h2').text());
```
Comment (limonte) [inline test/tests.js]: during fixing tests please take a look at https://github.com/limonte/sweetalert2/commit/736723e179487bc4c63e112d0c5ab561e7c710c0 ad change order of params in `equal()` functions - actual should be first param and expected should be second.

**Comment by simo9000** [comment]
Ok now it's failing locally. IDK why it wasn't before.

**Comment by limonte** [comment]
Nicely done @simo9000 ! :beers:

**Pull request merged**